src/Core/Services/Implementations/OrganizationService.cs

```diff
@@ -1681,14 +1681,14 @@ public async Task DeleteUserAsync(Guid organizationId, Guid userId)
         return result;
     }
 
-    public async Task<bool> HasConfirmedOwnersExceptAsync(Guid organizationId, IEnumerable<Guid> organizationUsersId, bool includeProvider = true)
+    public async Task<bool> HasConfirmedOwnersExceptAsync(Guid organizationId, IEnumerable<Guid> organizationUsersId, bool includeProvider = true, EventSystemUser? systemUser = null)
     {
         var confirmedOwners = await GetConfirmedOwnersAsync(organizationId);
         var confirmedOwnersIds = confirmedOwners.Select(u => u.Id);
         bool hasOtherOwner = confirmedOwnersIds.Except(organizationUsersId).Any();
         if (!hasOtherOwner && includeProvider)
         {
-            return (await _currentContext.ProviderIdForOrg(organizationId)).HasValue;
+            return systemUser == EventSystemUser.SCIM ? await _providerOrganizationRepository.GetByOrganizationId(organizationId) != null : (await _currentContext.ProviderIdForOrg(organizationId)).HasValue;
```

Comment:
I think the `systemUser` param is unnecessary - the `includeProvider` parameter should handle this situation, right? The idea is, "this org must have another owner OR a provider", which is exactly the case here (i.e. where a provider has set up the org and is using SCIM to sync users).

I think the bug is that `_currentContext.ProviderIdForOrg` is limited to `ProviderOrganization`s which the current user is a `ProviderUser` for. That's fine when the user is logging in via the provider portal and inviting users (because the user is a `ProviderUser`) but not for SCIM (which isn't a `ProviderUser`). We came across a similar issue in your reseller work as well.

The basic idea behind this method (as far as I can tell) is that we always want to make sure that an org has someone with owner-level access. That means either:
* an owner (other than the current user, if any)
* a confirmed ProviderUser (note that just checking for a Provider isn't enough, you want to make sure there is a `ProviderUser` who can actually log on)

So I don't think you need to worry about `systemUser` or exceptions for SCIM, just fix the logic to work as intended. Check for all `ProviderUsers` for this org, not just the current user's `ProviderUser`.